src/Features/Core/Portable/Debugging/DebugDataTipInfo.cs

```diff
@@ -6,10 +6,19 @@
 
 namespace Microsoft.CodeAnalysis.Debugging;
 
-internal readonly struct DebugDataTipInfo(TextSpan span, string text)
+internal enum DebugDataTipInfoKind
 {
-    public readonly TextSpan Span = span;
-    public readonly string Text = text;
+    None,
+    LinqExpression,
+}
+
+internal readonly record struct DebugDataTipInfo(
+    TextSpan Span, TextSpan ExpressionSpan, string? Text, DebugDataTipInfoKind Kind = DebugDataTipInfoKind.None)
```

Comment:
>TextSpan Span [](http://example.com/codeflow?start=4&length=13)

Have we implemented this such that if the cursor in anywhere in Span, the same expression to evaluate will be produced?

Reply:
Yes. That should be the semantics of 'span'.